app/models/apple/show.rb

```diff
@@ -8,18 +8,29 @@ class Show
       :private_feed,
       :api
 
+    def self.apple_shows_json(api)
+      api.get_paged_collection("shows")
+    end
+
     def self.apple_episode_json(api, show_id)
       api.get_paged_collection("shows/#{show_id}/episodes")
     end
 
     def self.connect_existing(apple_show_id, apple_config)
-      api = Apple::Api.from_apple_config(apple_config)
+      if (sl = SyncLog.find_by(feeder_id: apple_config.public_feed.id, feeder_type: :feeds))
+        if apple_show_id.blank?
+          sl.destroy!
+        elsif sl.external_id != apple_show_id
+          sl.update(external_id: apple_show_id)
+        end
+      end
```

Comment:
@svevang now it handles updated and deletes

Reply:
1. this is being called by the feed being updated with the apple id, so that would be circular if we also set the feed from this.

2. agreed, good point, I can add setting this on existing feeds from the sync log in the migration (or a new migration)

Reply:
added a routine to the migration!